### ISSUE TITLE
BiciMAD (Madrid) need an accessToken

### ISSUE THREAD

**Issue by Manu1400** [eskerda/pybikes#524]
## actual code
* https://github.com/eskerda/pybikes/blob/master/pybikes/bicimad.py
* https://github.com/eskerda/pybikes/blob/master/pybikes/data/bicimad.json

## source
https://gitlab.com/mobilitylabsmadrid/helps_and_utilities/openapi_documents/-/blob/master/utilities/DEVELOPERS.postman_collection.json#L580-L581

## documentation: other issue about BiciMAD
https://github.com/eskerda/pybikes/issues/428

**Comment by eUgEntOptIc44** [comment]
Hi @Manu1400 @eskerda 

*TL;DR* It took me quite a while but I managed to fix this. -> please see https://github.com/eUgEntOptIc44/pybikes/commit/66efe063d24a300ebfcf437b8dfec65a2711a90b

Unfortunately adding the `accessToken` mentioned by @Manu1400 didn't worked for me. It seems to be a placeholder or leftover from testing with *Postman* while documenting the api. Anyways the authentication fails and a error is returned.

So I dug deeper to discover that a account is required to generate a token. The token is only valid during the session. Furthermore a valid (Spanish?) mobile number is required to verify the account. All in all bad news for a possible integration into *pybikes*.

Searching the ['news'](https://mobilitylabs.emtmadrid.es/en/portal/noticias) section of Madrid's *Mobility Labs* I discovered a [map](https://mynavega.emtmadrid.es/) dashboard powered by *ArcGIS* at [mynavega.emtmadrid.es](https://mynavega.emtmadrid.es/). Despite the slightly dated impression it's fully functional and using some sort of proxy/loadbalancer to use the api data endpoints.

So using `https://mynavega.emtmadrid.es/services/api/BiciMadrid/GetAllStationsInfo` with the `referrer` header set to `https://mynavega.emtmadrid.es/?locale=es` works to get the bike sharing data. 🥳 

However I had to implement some minor adaptions to access the json encoded results within the json encoded request answer. This seems to be caused by the mentioned proxy/loadbalancer used for the map.

**Comment by eskerda** [comment]
> So I dug deeper to discover that a account is required to generate a token. The token is only valid during the session. Furthermore a valid (Spanish?) mobile number is required to verify the account. All in all bad news for a possible integration into pybikes.

We accept implementations requiring an access token, as long as this access token is something that can be reasonably accessed. See https://github.com/eskerda/pybikes/blob/master/pybikes/cyclocity.py for an example

I will dug deeper into their android app to see if there's any shortcut. While not ideal, I think it's reasonable to expect a token from https://mobilitylabs.emtmadrid.es/es/portal/mobilitylabs
